models/inventory/StockTransferItem.ts

```diff
@@ -215,6 +216,7 @@ export class StockTransferItem extends Doc {
 
   override hidden: HiddenMap = {
     batch: () => !this.fyo.singles.InventorySettings?.enableBatches,
+    serialNo: () => !this.fyo.singles.InventorySettings?.enableSerialNo,
```

Comment:
Should be visible only if the selected item has Serial Nos enabled (same with batches).